scripts/nbd_client_manager.py

```diff
@@ -32,19 +32,26 @@ def __init__(self, path):
         self._path = path
         self._lock_file = None
 
-    def lock(self):
+    def _lock(self):
         """Acquire the lock"""
         flags = fcntl.LOCK_EX
         self._lock_file = open(self._path, 'w+')
         fcntl.flock(self._lock_file, flags)
 
-    def unlock(self):
+    def _unlock(self):
         """Unlock and remove the lock file"""
         if self._lock_file:
             fcntl.flock(self._lock_file, fcntl.LOCK_UN)
             self._lock_file.close()
             self._lock_file = None
 
+    def __enter__(self):
+        self._lock()
+        return self
```

Comment:
You can remove the `return` if you don't need to do anything to the lock, that is optional anyway

Reply:
Thanks, done